testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java

```diff
@@ -73,15 +78,43 @@
     unFinal();
   }
 
-  static Starter stargateStarter;
-  public static String stargateHost = "127.0.0.11";
   public boolean enableAuth;
+
   protected final ClusterConnectionInfo backend;
 
+  public static List<Starter> stargateStarters = new ArrayList<>();
+  private static List<String> stargateHosts = new ArrayList<>();
+  public static final Integer numberOfStargateNodes = 3;
+
+  static {
+    for (int i = 1; i <= numberOfStargateNodes; i++) {
+      stargateHosts.add("127.0.0.1" + i);
+    }
+  }
+
   public BaseOsgiIntegrationTest(ClusterConnectionInfo backend) {
     this.backend = backend;
   }
 
+  public static String getStargateHost() {
+    return getStargateHost(0);
+  }
+
+  public static String getStargateHost(int stargateInstanceNumber) {
+    return stargateHosts.get(stargateInstanceNumber);
+  }
+
+  public static List<String> getStargateHosts() {
+    return stargateHosts;
+  }
+
+  public static List<InetAddress> getStargateInetSocketAddresses() throws UnknownHostException {
+    return Arrays.asList(
```

Comment:
minor: This is fragile if the number of stargate nodes is changed. Maybe iterate and convert instead?

```java
    return stargateHosts.stream().map(h -> {
      try {
        return InetAddress.getByName(h);
      } catch (UnknownHostException e) {
        throw new RuntimeException(e);
      }
    }).collect(Collectors.toList());
```

Reply:
good catch, done